cura/CuraPackageManager.py

```diff
@@ -99,64 +113,66 @@ def getInstalledPackageInfo(self, package_id: str) -> Optional[dict]:
             if package_id in Application.getInstance().getRequiredPlugins():
                 continue
+
+            # Add bundled plugins
+            if package_id in self._bundled_package_dict:
+                package_info = self._bundled_package_dict[package_id]["package_info"]
+
+            # Add installed plugins
+            if package_id in self._installed_package_dict:
+                package_info = self._installed_package_dict[package_id]["package_info"]
+
+            # Add to install plugins
             if package_id in self._to_install_package_dict:
                 package_info = self._to_install_package_dict[package_id]["package_info"]
-            else:
-                package_info = self._installed_package_dict[package_id]
-            package_info["is_bundled"] = False
-
-            package_type = package_info["package_type"]
-            if package_type not in installed_packages_dict:
-                installed_packages_dict[package_type] = []
-            installed_packages_dict[package_type].append( package_info )
 
             # We also need to get information from the plugin registry such as if a plugin is active
-            package_info["is_active"] = self._plugin_registry.isActivePlugin(package_id)
-
-        # Also get all bundled plugins
-        all_metadata = self._plugin_registry.getAllMetaData()
-        for item in all_metadata:
-            if item == {}:
-                continue
+            if package_info["package_type"] == "plugin":
+                package_info["is_active"] = self._plugin_registry.isActivePlugin(package_id)
+            else:
+                package_info["is_active"] = self._plugin_registry.isActivePlugin(package_id)
 
-            plugin_package_info = self.__convertPluginMetadataToPackageMetadata(item)
-            # Only gather the bundled plugins here.
-            package_id = plugin_package_info["package_id"]
-            if package_id in managed_package_id_set:
-                continue
-            if package_id in Application.getInstance().getRequiredPlugins():
-                continue
+            # If the package ID is in bundled, label it as such
+            if package_info["package_id"] in self._bundled_package_dict.keys():
+                package_info["is_bundled"] = True
+            else:
+                package_info["is_bundled"] = False
 
-            plugin_package_info["is_bundled"] = True if plugin_package_info["author"]["display_name"] == "Ultimaker B.V." else False
-            plugin_package_info["is_active"] = self._plugin_registry.isActivePlugin(package_id)
-            package_type = "plugin"
-            if package_type not in installed_packages_dict:
-                installed_packages_dict[package_type] = []
-            installed_packages_dict[package_type].append( plugin_package_info )
+            # If there is not a section in the dict for this type, add it
+            if package_info["package_type"] not in installed_packages_dict:
```

Comment:
When using defaultdict, you don't need this construction. But is it better? Don't know.